test/e2e/persistent_volumes.go

```diff
@@ -41,110 +58,155 @@ func nfsServerPodCleanup(c *client.Client, config VolumeTestConfig) {
 	if config.serverImage != "" {
 		podName := config.prefix + "-server"
 		err := podClient.Delete(podName, nil)
-		if err != nil {
-			framework.Logf("Delete of %v pod failed: %v", podName, err)
-		}
+		Expect(err).NotTo(HaveOccurred())
 	}
 }
 
-// Delete the PV. Fail test if delete fails. If success the returned PV should
-// be nil, which prevents the AfterEach from attempting to delete it.
-func deletePersistentVolume(c *client.Client, pv *api.PersistentVolume) (*api.PersistentVolume, error) {
+// Cleanup up pvs and pvcs in multi-pv-pvc test cases. All entries found in the pv and
+// claims maps are deleted.
+// Note: this is the only code that deletes PV objects.
+func pvPvcCleanup(c *client.Client, ns string, pvols pvmap, claims pvcmap) {
+
+	if c != nil && len(ns) > 0 {
+		for pvcKey := range claims {
+			_, err := c.PersistentVolumeClaims(pvcKey.Namespace).Get(pvcKey.Name)
+			if !apierrs.IsNotFound(err) {
+				Expect(err).NotTo(HaveOccurred())
+				framework.Logf("   deleting PVC %v ...", pvcKey)
+				//err = c.PersistentVolumeClaims(pvcKey.Namespace).Delete(pvcKey.Name)
+				Expect(err).NotTo(HaveOccurred())
+				framework.Logf("   deleted  PVC %v", pvcKey)
+			}
+			delete(claims, pvcKey)
+		}
 
-	if pv == nil {
-		return nil, fmt.Errorf("PV to be deleted is nil")
+		for name := range pvols {
+			_, err := c.PersistentVolumes().Get(name)
+			if !apierrs.IsNotFound(err) {
+				Expect(err).NotTo(HaveOccurred())
+				framework.Logf("   deleting PV %v ...", name)
+				err = c.PersistentVolumes().Delete(name)
+				Expect(err).NotTo(HaveOccurred())
+				framework.Logf("   deleted  PV %v", name)
+			}
+			delete(pvols, name)
+		}
 	}
+}
+
+// Delete the PV.
+func deletePersistentVolume(c *client.Client, pv *api.PersistentVolume) {
```

Comment:
Another good catch! This was code that was used at one point and eventually replaced.